pkg/callback/scheduler_callback.go

```diff
@@ -143,3 +145,48 @@ func (callback *AsyncRMCallback) ReSyncSchedulerCache(args *si.ReSyncSchedulerCa
 	}
 	return nil
 }
+
+// this callback implement scheduler plugin interface EventPlugin.
+func (callback *AsyncRMCallback) SendEvent(eventMessages []*si.EventMessage) error {
+	errors := make([]string, 0)
+	if len(eventMessages) > 0 {
+		log.Logger.Debug(fmt.Sprintf("processing %d events", len(eventMessages)))
+		for _, event := range eventMessages {
```

Comment:
For better modulization. I propose to add a publishEvent function for `task.go` (and similarly we can add for `application.go` or `node.go` in the future). So we can simply find out the task/app/node, and call `publishEvent(eventMessage *si.EventMessage)`

Reply:
Done.